src/cluster.c

```diff
@@ -5868,7 +6114,12 @@ clusterNode *getNodeByQuery(client *c, struct redisCommand *cmd, robj **argv, in
     /* Cluster is globally down but we got keys? We only serve the request
      * if it is a read command and when allow_reads_when_down is enabled. */
     if (server.cluster->state != CLUSTER_OK) {
-        if (!server.cluster_allow_reads_when_down) {
+        if (is_pubsublocal) {
+            if (!server.cluster_allow_pubsub_when_down) {
```

Comment:
My understanding is that today we still serve publish/subscribe even when the cluster is no ok? Not sure it really makes sense to change that behavior here?

Reply:
I thought we wanted to introduce a flag to let the user decide the behaviour.